ckanext/iati_generator/blueprint/iati.py

```diff
@@ -21,3 +23,37 @@ def iati_page(package_id):
         pkg=pkg_dict,
         pkg_dict=pkg_dict
     )
+
+
+@iati_blueprint.route("/<package_id>/generate", methods=["POST"])
+@require_sysadmin_user
+def generate_test_iati(package_id):
+    context = {"user": toolkit.c.user}
+    resource_id = request.form.get("resource_id")
+
+    if not resource_id:
+        flash("Resource ID is required", "error")
+        return redirect(url_for("iati_generator.iati_page", package_id=package_id))
+
+    result = toolkit.get_action("iati_generate_test_xml")(context, {"resource_id": resource_id})
+    logs = result.get("logs", "")
+
+    # Verificamos si se generó correctamente el archivo
+    xml_path = result.get("file_path")
+    if not xml_path:
+        flash("No se pudo generar el archivo XML. Revisá los logs abajo.", "error")
+
+    return render_template(
+        "package/iati_page.html",
+        pkg=toolkit.get_action("package_show")(context, {"id": package_id}),
+        pkg_dict=toolkit.get_action("package_show")(context, {"id": package_id}),
+        logs=logs,
```

Comment:
Send `result` to the template assuming it includes all relevant information about the process (even if they were errors)